include/pybind11/eigen/tensor.h

```diff
@@ -164,6 +164,8 @@ PYBIND11_WARNING_POP
 
 template <typename Type>
 struct type_caster<Type, typename eigen_tensor_helper<Type>::ValidType> {
+    static_assert(!std::is_pointer<typename Type::Scalar>::value,
```

Comment:
Any reason to not add the assert to eigen_tensor_helper instead to avoid duplicate lines?

Reply:
See above.